src/Foundation/ContainerUtil.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Foundation;
+
+class ContainerUtil
+{
+    public static function wrapCallback($callback, $container)
+    {
+        if (is_string($callback)) {
+            $callback = function () use ($container, $callback) {
+                $callback = $container->make($callback);
+
+                return call_user_func_array($callback, func_get_args());
+            };
```

Comment:
why not just  `$callback = $container->make($callback);` ?

Reply:
Clark pointed out in https://github.com/flarum/core/pull/2176#discussion_r517034525 that `$callback = $container->make($callback);` means that the callback will be resolved every time the site boots, even if its not needed for that request.

Reply:
my bad, didn't see that..

Reply:
I do agree it's a bit ugly, if there are simpler ways to do it, I'm all ears. But at least with the util, it's abstracted away so if we need to change it, that should be easy.

Reply:
it looks good to me :shrug: